sch_cake.c

```diff
@@ -875,12 +875,14 @@ static struct sk_buff *cake_ack_filter(struct cake_sched_data *q,
 	ipv6h = skb->encapsulation ? inner_ipv6_hdr(skb) : ipv6_hdr(skb);
 
 	/* check that the innermost network header is v4/v6, and contains TCP */
-	if (iph->version == 4) {
+	if (pskb_may_pull(skb, ((unsigned char *)iph - skb->head) + sizeof(struct iphdr)) &&
+	    iph->version == 4) {
```

Comment:
Maybe reverse these so logic short-circuiting dosn't call pskb_may_pull if it's not ipv4? (and likewise below)

Reply:
I’m not sure if there’s a case where there may not be any transport header, which the current order would defend against.

Reply:
Ah right, of course. Silly me. That makes me wonder if the encapsulation check above could be problematic?